### PR TITLE
Remove non-staking bakers

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
- **Remove bakers who dont stake, from the list of bakers in delegate and change baker list dialogue.**
- **Remove non-staking bakers from choose baker modal**

### CODE REVIEW

File: components/Operations/tezConnections.ts [hunk 1]
```diff
@@ -77,18 +79,32 @@ export function updateStakingOpsStatus(
...
+  //    A baker with nonzero balance or
+  //    a user which
+  //        has delegated to a baker and
+  //        has non-zero balance and
+  //        the baker who accepts staking and
+  //        has no pending unstake operations with another baker.
+
+  const { type, delegate, balance, stakedBalance } = accountInfo ?? {}
+  opStatus.Delegated =
+    (type === 'user' && Boolean(delegate)) || type === 'delegate'
```
Comment (rickykuo666) [inline components/Operations/tezConnections.ts:101]: can we use `!!delegate`?

File: components/Operations/tezConnections.ts [hunk 2]
```diff
@@ -113,11 +129,29 @@ export function updateStakingOpsStatus(
...
+    opStatus.bakerAcceptsStaking =
+      (Boolean(delegate) &&
+        bakerList?.find(baker => baker.address === delegate.address)
+          ?.acceptsStaking) ??
+      false
   }
 
+  if (type === 'delegate') {
+    opStatus.pendingUnstakeOpsWithAnotherBaker =
+      (prevBakerAddress && prevBakerAddress !== accountInfo.address) ?? false
```
Comment (rickykuo666) [inline components/Operations/tezConnections.ts:147]: can simplfy to `prevBakerAddress && prevBakerAddress !== accountInfo.address`

File: components/Operations/tezConnections.ts [hunk 3]
```diff
@@ -113,11 +129,29 @@ export function updateStakingOpsStatus(
...
-    opStatus.CanStake =
-      opStatus.CanStake && prevBakerAddress === delegate.address
+
+  if (type === 'user') {
+    opStatus.pendingUnstakeOpsWithAnotherBaker =
+      (prevBakerAddress &&
+        (!Boolean(delegate) || prevBakerAddress !== delegate?.address)) ??
+      false
+    opStatus.bakerAcceptsStaking =
+      (Boolean(delegate) &&
```
Comment (rickykuo666) [inline components/Operations/tezConnections.ts:139]: !!delegate && !!bakerList?.find(baker => baker.address === delegate.address)?.acceptsStaking)

**Review by rickykuo666 — CHANGES_REQUESTED** [review]
Only few mior changes requested

File: components/Operations/tezConnections.ts (hunk 1 above)
Comment (ajinkyaraj-23) [reply components/Operations/tezConnections.ts:101]: I prefer Boolean. Its much readable.

File: components/Operations/tezConnections.ts (hunk 2 above)
Comment (ajinkyaraj-23) [reply components/Operations/tezConnections.ts:147]: It throws an error. But following worked
` Boolean(prevBakerAddress) && prevBakerAddress !== accountInfo.address`

File: components/Operations/tezConnections.ts (hunk 3 above)
Comment (ajinkyaraj-23) [reply components/Operations/tezConnections.ts:139]: Done except used Boolean in place of !!

**Review by rickykuo666 — APPROVED** [review]

**Pull request merged**